src/nix/flake.cc

```diff
@@ -381,6 +381,11 @@ struct CmdFlakeCheck : FlakeCommand
         auto checkModule = [&](const std::string & attrPath, Value & v, const PosIdx pos) {
             try {
                 state->forceValue(v, pos);
+                if (v.type() == nPath) {
+                    PathSet context;
+                    auto path = state->coerceToPath(pos, v, context);
+                    state->evalFile(path, v);
```

Comment:
Since `v` is a reference pointing into the flake, `evalFile` will mutate the flake attribute.
That could lead to surprising behavior down the road.
You could copy `v` into a temporary `Value` after forcing `v` and work with that instead of `v`.

Reply:
Made a pointer to avoid copying.

Reply:
Actually that was silly, copying `Value`s is cheap.